Document CWalletTx::mapValue entries and remove erase of nonexistent "version" entry.

ryanofsky opened the pull request.

ryanofsky committed:
[trivial] Add comment documenting CWalletTx::mapValue

MarcoFalke commented:
Concept ACK, guess this could go to doxygen as well:
https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#doxygen-comments

On Mon, Dec 12, 2016 at 5:42 PM, Russell Yanofsky <notifications@github.com>
wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/bitcoin/bitcoin/pull/9333
> Commit Summary
>
>    - [trivial] Add comment documenting CWalletTx::mapValue
>
> File Changes
>
>    - *M* src/wallet/wallet.h
>    <https://github.com/bitcoin/bitcoin/pull/9333/files#diff-0> (20)
>
> Patch Links:
>
>    - https://github.com/bitcoin/bitcoin/pull/9333.patch
>    - https://github.com/bitcoin/bitcoin/pull/9333.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/9333>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGGmvwyHw-ubxSXuULI1M21IhefCtm8Rks5rHXl8gaJpZM4LKwdn>
> .
>

ryanofsky commented:
Thanks, switched to doxygen comment.

sipa commented:
utACK

fanquake commented:
utACK

jonasschnelli reviewed (approved):
Thanks for doing that!
ACK 28acf493f3c7cd461635a12e99fc3782dfe082a1

paveljanik commented on src/wallet/wallet.h:
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
+     *     "comment", "to"   - comment strings provided to sendtoaddress,
+     *                         sendfrom, sendmany wallet RPCs
+     *     "from", "message" - obsolete fields that could be set in UI prior to
+     *                         2011 (removed 4d9b223)
```
4d9b223 is the commit hash? github's UI can't find it. If it is commit hash, please add by/in before the ID.

ryanofsky replied:
Yeah not sure about the github interface. `git log 4d9b223` does work though.

paveljanik commented on src/wallet/wallet.h:
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
+     *                         2011 (removed 4d9b223)
+     *
+     * The following keys are serialized in the wallet database, but shouldn't
+     * be read or written through the map (they will be temporarily be added and
```
be... be...

ryanofsky replied:
Thanks, fixed.

paveljanik commented on src/wallet/wallet.h:
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
+     * removed from the map during serialization):
+     *
+     *     "fromaccount"     - serialized strFromAccount value
+     *     "version"         - serialized nTimeSmart value
```
version.... nTimeSmart?

ryanofsky replied:
Oops, fixed.

paveljanik commented on src/wallet/wallet.h:
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
+     *     "n"               - serialized nOrderPos value
+     *     "timesmart"       - serialized nTimeSmart value
+     *     "spent"           - serialized vfSpent value that existed prior to
+     *                         2014 (removed 93a18a3)
```
removed by/in

ryanofsky replied:
Thanks, added.

paveljanik commented:
ACK 654e044

laanwj commented on src/wallet/wallet.h:
```diff
@@ -248,6 +248,31 @@ class CWalletTx : public CMerkleTx
+     *     "spent"           - serialized vfSpent value that existed prior to
+     *                         2014 (removed in commit 93a18a3)
+     *
+     * A mapValue.erase("version") statement also appears in the code,
```
utACK - though would prefer to just remove this unnecessary code instead of bother to document it here

ryanofsky replied:
Ok, added another commit to this PR which removes the mapValue.erase("version") statement and this part of the comment.

ryanofsky committed:
Remove reference to nonexistent "version" wallet transaction mapvalue field

This change removes a mapValue.erase("version") statement which deletes a
mapValue entry that never existed. The statement was mistakenly added in commit
865c3a23832e36d50cb873d38c976032b027b5d3 in 2010 and is harmless but confusing.

ryanofsky committed:
[trivial] Add comment documenting bumpfee mapValues

ryanofsky commented:
New commit 87ed396159974b403d0ac00cb793888174e2665c adds documentation for bumpfee mapvalue entries (added in #8456)
